sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -793,6 +794,20 @@ object ConstantFolding extends Rule[LogicalPlan] {
 }
 
 /**
+ * Removes literal repetitions from IN predicate
+ */
+object RemoveLiteralRepetitionFromIn extends Rule[LogicalPlan] {
```

Comment:
Thank you for review, @rxin .
1. Sure, I can merge this into OptimizeIn.
2. Also, it can be used for deterministic expressions.
I'm just here focus on literals. May I handle both cases?